src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -1116,6 +1124,24 @@ void G1CollectedHeap::shrink(size_t shrink_bytes) {
   _verifier->verify_region_sets_optional();
 }
 
+bool G1CollectedHeap::request_heap_shrink(size_t shrink_bytes) {
+  if (shrink_bytes == 0) {
+    return false;
+  }
+
+  // Fast path: if we are already at a safepoint (e.g. called from the
+  // GC service thread) just do the work directly.
+  if (SafepointSynchronize::is_at_safepoint()) {
+    shrink(shrink_bytes);
+    return true;                     // we *did* something
```

Comment:
(Full) Sentence length comments should follow sentence style: Initial capitalization and full stop at the end. Not sure if the emphasis should be kept, seems to be overused (and I do not think we use it a lot in G1 code).
(Please no markdown btw :) )